### PR TITLE
XD87 qmk info fixes

### PR DESCRIPTION
**Pull request by siano**
## Description
XD87 failed to pass qmk info check. I've fixed it and updated info.json for this keyboard.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [X] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/xd87/xd87.h [hunk 1]
```diff
@@ -42,14 +42,14 @@
...
-    K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,          K2E,   K2F,   K2G,   \
-    K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   KC_NO, K3D,                               \
-    K40,   KC_NO, K42,   K43,   K44,   K45,   K46,   K47,   K48,   K49,   K4A,   K4B,   K4C,   KC_NO,        KC_NO, K4F,   KC_NO, \
-    K50,   K51,   K52,                 KC_NO,        K58,                 K5A,   K5B,   K5C,   K5D,          K5E,   K5F,   K5G    \
-)
+) { \
+    {K00,   KC_NO, K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,          K0E,   K0F,   K0G}, \
+    {K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,          K1E,   K1F,   K1G}, \
+    {K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   K2D,          K2E,   K2F,   K2G}, \
+    {K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   KC_NO, K3D}, \
```
Comment (fauxpark) [inline keyboards/xd87/xd87.h:49]: Are you sure this works correctly? Each row should have `MATRIX_COLS` number of elements, with nonexistent matrix positions blanked out by `KC_NO`s.
Weirdly, Travis doesn't seem to mind, but perhaps GCC is just silently padding the truncated rows out.

Comment (noroadsleft) [reply keyboards/xd87/xd87.h:49]: > perhaps GCC is just silently padding the truncated rows out.

That's exactly what happens – GCC pads the truncated rows out with NULL, which ends up the same as using `KC_NO` (which is defined as `0x00` in `tmk_core/common/keycode.h`. This paradigm used to be all over the repo when I was starting out with contributing to QMK.

Comment (noroadsleft) [reply keyboards/xd87/xd87.h:49]: Back to add: As for the parts that skip entries in the middle of a row, I'm not convinced that works.

Comment (siano) [reply keyboards/xd87/xd87.h:49]: The funny part is, that the bottom row is broken. I'm fixing it now.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**